_posts/2022-08-14-aug-update.markdown

```diff
@@ -0,0 +1,171 @@
+[our "Porting Software to Managarm" post](https://managarm.org/2020/04/05/porting-software.html),
+in 2020 we implemented pthread thread creation and other related functions. mlibc has also
+gotten a pthread cancellation implementation, and we have an upcoming blog post
+going into detail about it.
+
+**A new IDL Compiler: bragi.** In February 2020, we started work on our own interface description language called bragi. The aim is to replace all of our current protobuf usage with bragi. Although it is not yet fully feature-complete (we still want/need to add features like variants or inheritance), bragi is already mature enough to enable us to refactor some of our IPC protocols (namely, the POSIX, hardware, and filesystem protocols) to use bragi. We also started implementing new protocols (like the ostrace one) based on bragi.
+
+**New Drivers: AHCI and NVMe and Storage Improvements.** Managarm's block driver stack received significant updates in 2021. We now have drivers for the two most important modern block device controllers AHCI and NVMe. The AHCI driver was written by Matt Taylor ([@64](https://github.com/64)), while the NVMe driver was contributed by Jin Xue ([@Jimx-](https://github.com/Jimx-)). Furthermore, due to a PR by Geert Custers ([@geertiebear](https://github.com/geertiebear)), we can now identify partitions by their UUIDs. This feature will make identifying the boot device more robust in the future and is especially important when running Managarm on physical hardware (and not in a virtual machine).
+
+**Networking Improvements.** Our networking stack ("netserver") can now connect to TCP servers over IPv4. It supports basic TCP features; however, the server side of the TCP 3-way handshake and path MTU discovery is not implemented yet. Once these gaps are filled, we will have a mostly complete IPv4 stack.
+
+## New Ports and Port Updates
+
+In the last two years, we received a lot of new and sometimes updated ports, our collection contains over 250 ports now! A lot of the ports are various nice-to-have things, such as common UNIX utilities like `grep`, `sed`, `findutils` and `gawk`, development tools like `python`, `make` and `patch` and we got enough of the X11 stack ported to run XWayland and several X based apps like `xclock` and `gtklife`. Another noteworthy thing to mention here is the addition of a new bootloader called [limine](https://limine-bootloader.org/), which we now use by default (although `grub` is still supported at this time and there are no plans to remove that support) and the addition of a stripped down `util-linux` port, which includes useful utilities like `mount` and `losetup`.
+
+As a blog post without images would be boring, here are some screenshots, first off is Managarm running `python`.
+![python](/assets/2022-aug-update/python.png)
+After that, we have `xclock`.
+![xclock](/assets/2022-aug-update/xclock.png)
+And finally we have `exa` running.
+![exa](/assets/2022-aug-update/exa.png)
```

Comment:
Perhaps @Dennisbonke wants to add a screenshot of chromium here?

Reply:
I think we only want to show case things that are upstreamed here.

Reply:
And before anyone mentions it, Qt6 will be discussed in the next update post (end of year 2022 is the planning). And qookie is right, only upstreamed items are mentioned.